arch/arm/include/kernel_arch_func.h

```diff
@@ -97,7 +97,9 @@ _arch_switch_to_main_thread(struct k_thread *main_thread,
 		*/
 		"mov %%r0, %3 \t\n"
 		"push {r2, lr} \t\n"
+		"push {r3, lr} \t\n"
```

Comment:
I have to wonder if we also need this in the pendsv code.  We do a push {r2, lr} there.

Reply:
This is actually just a symptom of a larger problem which is consistent saving/restoring of caller/callee registers during subroutine calls from within assembly routines.